magefiles/rulesengine/repos/e2e_repo.go

```diff
@@ -16,81 +17,295 @@ var NonTestFilesRule = rulesengine.Rule{Name: "E2E Default PR Test Exectuion",
 
-}
+		}
 
-func CheckMageFilesChanged(rctx *rulesengine.RuleCtx) bool {
+		return nil
+
+	})}}
+
+var BuildNonTestFileChangeRule = rulesengine.Rule{Name: "E2E PR Build Test Helper Files Change Rule",
+	Description: "Map build tests files when const.go or source_build.go file is changed in the e2e-repo PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("tests/build/const.go")) != 0 || len(rctx.DiffFiles.FilterByDirGlob("tests/build/source_build.go")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
 
-	for _, file := range rctx.DiffFiles {
+		for _, file := range rctx.DiffFiles.FilterByDirGlob("tests/build/*.go") {
+
+			if strings.Contains(file.Name, "source_build.go") || strings.Contains(file.Name, "const.go") || strings.Contains(file.Name, "scenarios.go") {
+
+				continue
+
+			}
+			rctx.FocusFiles = dedupeAppendFiles(rctx.FocusFiles, file.Name)
 
-		switch {
-		case strings.Contains(file.Name, "magefile/"):
-			return true
 		}
 
-	}
+		return nil
 
-	return false
+	})}}
 
-}
+var ReleaseTestTestFilesChangeRule = rulesengine.Rule{Name: "E2E PR Release Test File Change Rule",
+	Description: "Map release test files if they are changed in the e2e-repo PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
 
-func CheckCmdFilesChanged(rctx *rulesengine.RuleCtx) bool {
+		return len(rctx.DiffFiles.FilterByDirGlob("tests/release/*/*.go")) != 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
 
-	for _, file := range rctx.DiffFiles {
+		for _, file := range rctx.DiffFiles.FilterByDirGlob("tests/release/*/*.go") {
 
-		switch {
+			rctx.FocusFiles = dedupeAppendFiles(rctx.FocusFiles, file.Name)
 
-		case strings.Contains(file.Name, "cmd/"):
-			return true
 		}
 
-	}
+		return nil
+
+	})}}
+
+var ReleaseTestHelperFilesChangeOnlyRule = rulesengine.Rule{Name: "E2E PR Release Test Helper File CHange Rule",
+	Description: "Map release tests files when only the release helper go files in root of release directory are changed in the e2e-repo PR",
+	Condition: rulesengine.ConditionFunc(func(rctx *rulesengine.RuleCtx) bool {
+
+		return len(rctx.DiffFiles.FilterByDirGlob("tests/release/*.go")) != 0 && len(rctx.DiffFiles.FilterByDirGlob("tests/release/*/*.go")) == 0
+	}),
+	Actions: []rulesengine.Action{rulesengine.ActionFunc(func(rctx *rulesengine.RuleCtx) error {
+
+		matched, err := filepath.Glob("tests/release/*/*.go")
+		if err != nil {
+
+			return err
+		}
+		for _, matched := range matched {
+
+			rctx.FocusFiles = dedupeAppendFiles(rctx.FocusFiles, matched)
+		}
+
+		return nil
+
+	})}}
+
+var ReleaseConstFileChangeBuildTestRule = rulesengine.Rule{Name: "E2E PR Release Test Const File Dependency Change Rule",
+	Description: "Map build tests files when release const.go file is changed in the e2e-repo PR",
```

Comment:
is there co-relation between release `const.go`  and `tests/build/build.go` and when this rule will be applicable?

Reply:
Yeah, from what I see in build.go it has an import located here. https://github.com/konflux-ci/e2e-tests/blob/3bf858442226b3442279086d80cf3b3b9ab0bd4d/tests/build/build.go#L29

It looks like its primarily a dependency on the const.go file within release test directory. So the idea is if const.go in release directory is modified. We should add the build.go test file. That was the idea. 

Let me know if that is correct or if you think it should be written in a different way to make the rule clearer?

Reply:
Okay understood it now... since these [two variables](https://github.com/konflux-ci/e2e-tests/blob/3f5c75d639864f121f836391c98ad99b17a353b6/tests/release/const.go#L61-L62) used in both release and build test, I was thinking if it would good to move them to  [global constants pkg](https://github.com/konflux-ci/e2e-tests/blob/main/pkg/constants/constants.go) and modify the rule accordingly

Reply:
Yes, I was thinking similar! I didn't really like the cross-depedency but I didn't want to touch anything with consulting with you guys. @jinqi7 thoughts on this? Would you be able to validate with the release team?

Reply:
I think they can be moved to global constants pkg. But currently it works also in the PR, right? Maybe change it later?

Reply:
I'm fine with addressing later in another PR if everyone is ok?

Reply:
Okay with addressing that with another PR.. without this change being merged we are not able to validate [this PR](https://github.com/konflux-ci/e2e-tests/pull/1273) changes in CI